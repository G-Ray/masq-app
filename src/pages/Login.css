.Login h1 {
  font-size: 30px;
  line-height: 0.83;
  text-align: center;
  color: #191919;
}

.avatars-grid .avatar {
  background-image: url('https://randomuser.me/api/portraits/women/79.jpg');
  width: 120px;
  height: 120px;
  min-width: 120px;
  background-size: cover;
  background-position: top center;
  border-radius: 50%;
}

.avatars-grid .add-avatar {
  width: 120px;
  height: 120px;
  min-width: 120px;
  background-size: cover;
  background-position: top center;
  border-radius: 50%;
}

.avatars-grid p {
  text-align: center;
  font-size: 18px;
  font-weight: 600;
  line-height: 1.11;
  color: #4c4c4c;
}

.avatars-grid {
  display: grid;
  grid-template-columns: auto auto;
  grid-column-gap: 40px;
  grid-row-gap: 15px;
  justify-content: center;
}
