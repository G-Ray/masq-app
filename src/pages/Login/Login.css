.Login h1 {
  font-size: 30px;
  line-height: 0.83;
  text-align: center;
  color: #191919;
}

.Login .avatars-grid .avatar {
  background-image: url('https://randomuser.me/api/portraits/women/79.jpg');
  width: 120px;
  height: 120px;
  min-width: 120px;
  background-size: cover;
  background-position: top center;
  border-radius: 50%;
}

.Login .avatars-grid .add-avatar {
  width: 120px;
  height: 120px;
  min-width: 120px;
  background-size: cover;
  background-position: top center;
  border-radius: 50%;
}

.Login .avatars-grid p {
  text-align: center;
  font-size: 18px;
  font-weight: 600;
  line-height: 1.11;
  color: #4c4c4c;
}

.Login .avatars-grid {
  display: grid;
  grid-template-columns: auto auto;
  grid-column-gap: 40px;
  grid-row-gap: 15px;
  justify-content: center;
}
