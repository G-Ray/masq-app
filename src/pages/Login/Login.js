import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { Plusbutton } from 'icons'
import { Avatar } from 'components'

import './Login.css'

export default function Login (props) {
  const { users, auth } = props
  const style = !users.length ? { display: 'flex' } : {}
  return (
    <div className='Login'>
      <h1>Who is it ?</h1>

      <div className='avatars-grid' style={style}>
        {users.map((user, index) =>
          <Link style={{textDecoration: 'none'}} key={index} to='devices' onClick={() => auth(index)}>
            <Avatar image={user.image} />
            <p>{user.username}</p>
          </Link>
        )}

        <Link to='register' style={{textDecoration: 'none'}}>
          <div style={{height: '120px'}}>
            <Plusbutton />
          </div>
          <p id='add-user'>Add user</p>
        </Link>
      </div>
    </div>
  )
}

Login.propTypes = {
  auth: PropTypes.func.isRequired,
  users: PropTypes.array.isRequired
}
