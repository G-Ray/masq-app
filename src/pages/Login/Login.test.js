import React from 'react'
import ReactDOM from 'react-dom'
import Login from './Login'
import { MemoryRouter } from 'react-router'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<MemoryRouter><Login users={[]} auth={() => true} /></MemoryRouter>, div)
})
