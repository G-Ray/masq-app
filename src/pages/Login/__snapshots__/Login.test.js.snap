// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login with users 1`] = `
<div
  className="Login"
>
  <h1>
    Who is it ?
  </h1>
  <div
    className="avatars-grid"
    style={Object {}}
  >
    <a
      href="devices"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="Avatar"
        style={
          Object {
            "backgroundImage": "url(https://randomuser.me/api/portraits/women/79.jpg)",
          }
        }
      />
      <p>
        Mustang
      </p>
    </a>
    <a
      href="devices"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="Avatar"
        style={
          Object {
            "backgroundImage": "url(https://randomuser.me/api/portraits/men/1.jpg)",
          }
        }
      />
      <p>
        Jeff
      </p>
    </a>
    <a
      href="register"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        style={
          Object {
            "height": "120px",
          }
        }
      >
        <svg
          height={120}
          viewBox="0 0 120 120"
          width={120}
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <circle
              cx={60}
              cy={60}
              id="plus-button-svg-a"
              r={60}
            />
            <path
              d="M78.667 60c0 1.4-.934 2.333-2.334 2.333h-14v14c0 1.4-.933 2.334-2.333 2.334-1.4 0-2.333-.934-2.333-2.334v-14h-14c-1.4 0-2.334-.933-2.334-2.333 0-1.4.934-2.333 2.334-2.333h14v-14c0-1.4.933-2.334 2.333-2.334 1.4 0 2.333.934 2.333 2.334v14h14c1.4 0 2.334.933 2.334 2.333z"
              id="plus-button-svg-b"
            />
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <use
              fill="#FFF"
              xlinkHref="#plus-button-svg-a"
            />
            <circle
              cx={60}
              cy={60}
              r={59}
              stroke="#CCC"
              strokeWidth={2}
            />
            <use
              fill="#CCC"
              fillRule="nonzero"
              xlinkHref="#plus-button-svg-b"
            />
          </g>
        </svg>
      </div>
      <p
        id="add-user"
      >
        Add user
      </p>
    </a>
  </div>
</div>
`;

exports[`renders Login without users 1`] = `
<div
  className="Login"
>
  <h1>
    Who is it ?
  </h1>
  <div
    className="avatars-grid"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <a
      href="register"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        style={
          Object {
            "height": "120px",
          }
        }
      >
        <svg
          height={120}
          viewBox="0 0 120 120"
          width={120}
          xmlnsXlink="http://www.w3.org/1999/xlink"
        >
          <defs>
            <circle
              cx={60}
              cy={60}
              id="plus-button-svg-a"
              r={60}
            />
            <path
              d="M78.667 60c0 1.4-.934 2.333-2.334 2.333h-14v14c0 1.4-.933 2.334-2.333 2.334-1.4 0-2.333-.934-2.333-2.334v-14h-14c-1.4 0-2.334-.933-2.334-2.333 0-1.4.934-2.333 2.334-2.333h14v-14c0-1.4.933-2.334 2.333-2.334 1.4 0 2.333.934 2.333 2.334v14h14c1.4 0 2.334.933 2.334 2.333z"
              id="plus-button-svg-b"
            />
          </defs>
          <g
            fill="none"
            fillRule="evenodd"
          >
            <use
              fill="#FFF"
              xlinkHref="#plus-button-svg-a"
            />
            <circle
              cx={60}
              cy={60}
              r={59}
              stroke="#CCC"
              strokeWidth={2}
            />
            <use
              fill="#CCC"
              fillRule="nonzero"
              xlinkHref="#plus-button-svg-b"
            />
          </g>
        </svg>
      </div>
      <p
        id="add-user"
      >
        Add user
      </p>
    </a>
  </div>
</div>
`;
